shopinvader_anonymous_partner/models/res_partner.py

```diff
@@ -95,9 +47,11 @@ def _get_anonymous_partner__token(self, token: str):
             .search([("anonymous_token", "=", token)], limit=1)
         )
 
-    @api.model
-    def _get_anonymous_partner__cookie(self, cookies: Cookies):
-        token = cookies.get(COOKIE_NAME)
-        if not token:
-            return self.env["res.partner"].sudo().browse()
-        return self._get_anonymous_partner__token(token)
+    def _promote_from_anonymous_partner(self, anonymous_partner: ResPartner):
+        """
+        Promote an anonymous partner to an authenticated partner.
+
+        This method should be overridden by other modules to implement
+        the partner resolution logic, merging the anonymous partner cart
+        for instance.
+        """
```

Comment:
Would it make sense to raise a NotImplementedError here?

Reply:
No, as many module can implement a behaviour of promotion (the shopinvader_api_cart will assign the cart to the logged partner, a lead module can link a question to an logged partner...) so each module will call the super method